include/net/net_pkt.h

```diff
@@ -1222,6 +1222,8 @@ static inline struct net_buf *net_pkt_write_be32(struct net_pkt *pkt,
  * is based on fragment length (only user written data length, any tailroom
  * in fragments does not come to consideration unlike net_pkt_write()) and
  * calculates from input fragment starting position.
+ * If the data pointer is NULL, then the function will just insert an empty
+ * area in the buf where the values are set to 0.
```

Comment:
This isn't exactly clear. I'd reformulate and mentioned that number of inserted zeros is still given by `len` param. E.g. "If the data pointer is NULL, insert a sequence of zeros with the given length."

Reply:
indeed, rephrase this. "Empty area" is confusing. Filling in 0s does not make the area empty to me, depending on the use case (padding or whitening).

Something like that:
"If the data pointer is NULL, the function will fill in the area with only zeros instead."